test/camunda-cloud/CleanUpTaskListenersBehaviorSpec.js

```diff
@@ -0,0 +1,191 @@
+import {
+  bootstrapCamundaCloudModeler,
+  inject
+} from 'test/TestHelper';
+
+import { getExtensionElementsList } from 'lib/util/ExtensionElementsUtil';
+
+import { getBusinessObject } from 'bpmn-js/lib/util/ModelUtil';
+
+import diagramXML from './task-listeners.bpmn';
+
+
+describe('camunda-cloud/features/modeling - CleanUpTaskListenersBehavior', function() {
+
+  beforeEach(bootstrapCamundaCloudModeler(diagramXML));
+
+  describe('remove execution listeners if disallowed in element type', function() {
+
+    const testCases = [
+      {
+        title: 'User Task -> Start Event',
+        element: 'UserTask',
+        target: {
+          type: 'bpmn:StartEvent'
+        }
+      },
+      {
+        title: 'User Task -> Gateway',
+        element: 'UserTask',
+        target: {
+          type: 'bpmn:Gateway'
```

Comment:
I'd rather not use abstract types in tests as it is impossible to reproduce it in the editor:

```suggestion
        title: 'User Task -> ParallelGateway',
        element: 'UserTask',
        target: {
          type: 'bpmn:ParallelGateway'
```